src/CommonLib/hostfxr_utility.cpp

```diff
@@ -369,3 +342,166 @@ HOSTFXR_UTILITY::SetHostFxrArguments(
+    pwzDotnetName = SysAllocString(L"\"C:\\Windows\\System32\\where.exe\" dotnet.exe");
+
+    fResult = CreateProcessW(NULL,
+        pwzDotnetName,
+        NULL,
+        NULL,
+        TRUE,
+        CREATE_NO_WINDOW,
+        NULL,
+        NULL,
+        &startupInfo,
+        &processInformation
+    );
+
+    if (!fResult)
+    {
+        hr = ERROR_PROCESS_ABORTED;
+        goto Finished;
+    }
+
+    WaitForSingleObject(processInformation.hProcess, pConfig->QueryRequestTimeoutInMS());
+
+    fResult = GetExitCodeProcess(processInformation.hProcess, &dwExitCode);
+
+    // Don't check the result of Closehandle here
+    // Just a guarantee the process is closed.
+    CloseHandle(processInformation.hProcess);
+    CloseHandle(processInformation.hThread);
+
+    // Reset file pointer to the beginning of the file. 
+    dwFilePointer = SetFilePointer(hStdOutReadPipe, 0, NULL, FILE_BEGIN);
+    if (dwFilePointer == INVALID_SET_FILE_POINTER)
+    {
+        hr = ERROR_FILE_INVALID;
+        goto Finished;
+    }
+
+    if (!ReadFile(hStdOutReadPipe, pzFileContents, READ_BUFFER_SIZE, &dwNumBytesRead, NULL))
+    {
+        hr = ERROR_FILE_INVALID;
+        goto Finished;
+    }
+    if (dwNumBytesRead == READ_BUFFER_SIZE)
+    {
+        // This shouldn't ever be this large. We could continue to call ReadFile in a loop,
+        // however I'd rather error out here and report an issue.
+        hr = ERROR_FILE_TOO_LARGE;
+        goto Finished;
+    }
+    struDotnetLocationsString.CopyA(pzFileContents, dwNumBytesRead);
+
+    // Check the bitness of the currently running process
+    // matches the dotnet.exe found. 
+    if (!IsWow64Process(GetCurrentProcess(), &fIsWow64Process))
+    {
+        // Calling IsWow64Process failed
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        goto Finished;
+    }
+    if (fIsWow64Process)
+    {
+        // 32 bit mode
+        fIsCurrentProcess64Bit = FALSE;
+    }
+    else
+    {
+        SYSTEM_INFO systemInfo;
+        GetNativeSystemInfo(&systemInfo);
+        fIsCurrentProcess64Bit = systemInfo.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_AMD64;
+    }
+
+    while (!fFound)
+    {
+        index = struDotnetLocationsString.IndexOf(L"\r\n", prevIndex);
+        if (index == -1)
+        {
+            break;
+        }
+        struDotnetSubstring.Copy(struDotnetLocationsString.QueryStr(), index - prevIndex);
+        prevIndex = index;
+
+        if (!GetBinaryTypeW(struDotnetSubstring.QueryStr(), &dwBinaryType))
+        {
+            // TODO should we ignore this failure and continue to trying the next dotnet.exe?
+            hr = HRESULT_FROM_WIN32(GetLastError());
+            goto Finished;
+        }
+        if (fIsCurrentProcess64Bit == (dwBinaryType == SCS_64BIT_BINARY)) {
+            // Found a valid dotnet.
+            struDotnetPath->Copy(struDotnetSubstring);
+            fFound = TRUE;
+        }
+    }
+
+    if (!fFound)
+    {
+        hr = ERROR_FILE_NOT_FOUND;
```

Comment:
How can the user get visibility into this failure? E.g. how can they get a verbose trace of what paths were searched and why they failed?

Reply:
I was just about the comment on that. I have another PR: https://github.com/aspnet/IISIntegration/pull/533 which logs failures in hostfxr_utility. I was going to add logs throughout this call. I can rebase that branch onto this branch until the PR is in. 

To know which paths were searched, in the windows event log I will output the contents as part of the error message.